### PR TITLE
IBC Restructuring

### PR DESCRIPTION
**Pull request by chipshort**
merging into #136

After thinking a bit about the feedback in #121, I decided to split the IBC docs a little differently. I'm still not 100% happy with it, so suggestions are welcome.

I removed the paragraph about using IBC through another contract and instead focus on ICS20 on that page, also renaming it to that. I moved the IBC Callbacks part out of that into its own page under an "Extensions" menu point. This can also house a page about IBC Fees when they are implemented and I start writing about them.

### CODE REVIEW

**Review by dakom — APPROVED** [review]
Looks good to me! Makes it easier to navigate and focus on a specific topic :)

One small question/thought - it seems to me IBC callbacks are specifically for getting called back on IBC messages that the contract did _not_ initiate. If I understood that correctly, it might be worth adding a line like that somewhere towards the top.

**edit: this is reminiscent of the discussion here: https://vercel.live/link/cosmwasm-docs-qn1m4t060-confio.vercel.app?page=%2Fibc%2Fdiy-protocol%3FvercelThreadId%3D1Tpuh&via=in-app-copy-link - if you've already considered this in the PR, all good, just making sure it's on your radar**

In other words, at first glance, it isn't immediately clear to me why I'd want a callback vs. a regular ack handler. The example of ICS-20 does illustrate this, i.e. in the typical case, like via wallets, the message doesn't originate from contracts at all - but it takes a bit of figuring out to make that connection.

All in all, suggestion to add, something like:

> If you send an IBC message from a contract, you'll get an acknowledgement of receipt in the ack handler, but what if you want to get notified on completion of IBC messages you did not send? One real-world use-case of this is ICS-20 messages. They might have been sent from a wallet and not a contract at all, so how do you get notified in a contract when the transfer completes? Enter IBC-Callbacks...

Actually - that's a bit too wordy... but it's the best I could come up with right now 🤷  so... consider it just a light suggestion/idea, not a definite request for change :)

**Comment by chipshort** [comment]
Thanks David, that's a great point. I was initially wary to add it to the intro for callbacks, since they came before the explanation of ack and timeout for packets, so I thought it would be weird to refer to them when they weren't even explained yet. Since this is no longer the case, I will adjust it with your suggestion, as it is indeed an important point.
